### PR TITLE
Use jenkins-button for Repeatable buttons

### PR DESCRIPTION
**Pull request by janfaracik**
**Before**
image

**After**
image

### Testing done
* Repeatable buttons work as before, tried project configuration pages and tools page

### Proposed changelog entries

- Use jenkins-button for Repeatable buttons

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7717"></a>

### CODE REVIEW

File: core/src/main/resources/lib/form/repeatable/repeatable.js
Comment (janfaracik) [inline core/src/main/resources/lib/form/repeatable/repeatable.js:112]: Is there a potential this could be used for an XSS attack?

**Comment by timja** [comment]
can you run this through ATH please?

**Review by timja — APPROVED** [review]

File: core/src/main/resources/lib/form/repeatable/repeatable.js
Comment (yaroslavafenkin) [reply core/src/main/resources/lib/form/repeatable/repeatable.js:112]: I've tried to inject various things via `add` attribute and it didn't work out. Content was properly escaped, presumably because of `escape-by-default='true'`.

**Review by yaroslavafenkin — APPROVED** [review]
LGTM

**Comment by timja** [comment]
Tests failing fyi @janfaracik

**Comment by janfaracik** [comment]
Hey @timja, could you lend a hand with this if you have the time? The tests aren't making much sense to me

**Comment by timja** [comment]
> Hey @timja, could you lend a hand with this if you have the time? The tests aren't making much sense to me

Run the test in a debugger, add `j.pause()` somewhere in it.
Open the test url, e.g. for me it was:
http://localhost:51255/jenkins/self/testNestedRadioTopButton

The url of the Jenkins instance will be logged near the beginning of testing. Append `self/testNestedRadioTopButton` to the end of the URL.

It will look like this:
image

The test is making sure that the payload submitted with repeatable elements is as expected, e.g.

image


When you submit it you see the JSON after the first element is not correct:

```json
{"items": [{"moo": [{"inner": "inone"}, {"inner": "intwo"}]}, {"outer": "one", "moo": {"inner": "intwo"}}, {"outer": "two"}], "Submit": "", "Jenkins-Crumb": "test"}
```

Hope that helps, let me know if you need more of a hand.

File: test/src/test/java/lib/form/RepeatableTest.java
Comment (daniel-beck) [inline test/src/test/java/lib/form/RepeatableTest.java:602]:
```suggestion
```

Should not be merged as is.

**Comment by timja** [comment]
> 14:11:21  [ERROR] lib.form.RepeatableTest.testNestedEnabledTopButtonOuter  Time elapsed: 6.624 s  <<< ERROR!
14:11:21  java.lang.NullPointerException: Cannot read the array length because "this.list" is null
14:11:21  	at lib.form.RepeatableTest$FooList.toString(RepeatableTest.java:424)
14:11:21  	at java.base/java.lang.String.valueOf(String.java:4216)
14:11:21  	at java.base/java.lang.StringBuilder.append(StringBuilder.java:173)
14:11:21  	at java.base/java.util.AbstractCollection.toString(AbstractCollection.java:457)
14:11:21  	at lib.form.RepeatableTest.testNestedEnabledTopButtonOuter(RepeatableTest.java:546)
14:11:21  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
14:11:21  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
14:11:21  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:11:21  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)

**Comment by timja** [comment]
Nope not yet, the build log has this at the end:

```
19:13:58  java.lang.NullPointerException
19:13:59  [GitHub Checks] GitHub check (name: Jenkins, status: completed) has been published.
19:13:59  Finished: FAILURE
```

**Comment by timja** [comment]
> Any idea why the build is failing @timja?

Passing now

**Comment by timja** [comment]
This PR is currently pending merge and release of https://github.com/jenkinsci/scriptler-plugin/pull/59 to unblock ATH.

**Review by NotMyFault — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by basil** [comment]
Breaks `ExtendedEmailPublisherDescriptorTest#groovyClassPath` with

```
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at hudson.plugins.emailext.ExtendedEmailPublisherDescriptorTest.groovyClassPath(ExtendedEmailPublisherDescriptorTest.java:325)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:608)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by basil** [comment]
As a courtesy I have released Email Extension [2.98](https://github.com/jenkinsci/email-ext-plugin/releases/tag/email-ext-2.98).

**Pull request merged**
